tests/test_config.py

```diff
@@ -1,37 +1,98 @@
+# pylint: disable=missing-module-docstring, missing-function-docstring, protected-access
 import unittest.mock
 
 import pylint.lint
 
 
-def test_can_read_toml(tmp_path):
-    config_file = tmp_path / "pyproject.toml"
+def check_configuration_file_reader(config_file):
+    """Initialize pylint with the given configuration file and check that
+    what we initialized the linter with what was expected.
+    """
+    args = ["--rcfile", str(config_file), __file__]
+    # If we used `pytest.raises(SystemExit)`, the `runner` variable
+    # would not be accessible outside the `with` block.
+    with unittest.mock.patch("sys.exit") as mocked_exit:
+        # Do not actually run checks, that could be slow.
+        with unittest.mock.patch("pylint.lint.pylinter.PyLinter.check"):
+            runner = pylint.lint.Run(args)
+
+    enabled = {
+        msgid for msgid, _enabled in runner.linter._msgs_state.items() if _enabled
+    }
+    # logging-format-interpolation, logging-fstring-interpolation
+    assert enabled == {"W1202", "W1203"}
+    assert runner.linter.config.jobs == 10
+    type_checker = runner.linter._checkers["typecheck"][0]
```

Comment:
I would prefer if we'd not check this as it depends on a private variable set in a checker. The number of jobs though should be fine.